src/util/ByteOrder.hxx

```diff
@@ -516,7 +517,55 @@ public:
 			x = ByteSwap64(x);
 		return x;
 	}
-};
+} __attribute__((packed));
 
 static_assert(sizeof(PackedLE64) == sizeof(uint64_t), "Wrong size");
 static_assert(alignof(PackedLE64) == 1, "Wrong alignment");
+
+class PackedFloatLE
+{
+  union Value
+  {
+    uint32_t unsigned_int = 0;
+    float floating_point;
+  } __attribute__((packed)) value;
+
+public:
+  PackedFloatLE() = default;
+
+  PackedFloatLE(float f)
+  {
+    value.floating_point = f;
+    if (IsBigEndian()) {
+      value.unsigned_int = ByteSwap32(value.unsigned_int);
+    }
+  }
+} __attribute__((packed));
+
+static_assert(std::numeric_limits<float>::is_iec559);
+static_assert(sizeof(PackedFloatLE) == sizeof(uint32_t), "Wrong size");
```

Comment:
This is an obscure requirement. Why should the size of this class be the same as `uint32_t`? It's not a 32-bit integer. It's a 32-bit float. Sure, in the end, both are 4 bytes, but you're not testing for it to be 32 bit or 4 bytes, you're testing this to be uint32_t, but why this choice? This is confusing.

Reply:
I know for sure that the result (to be sent to the vario) should be 4 bytes. What if the float is not 4 bytes on some platform? Maybe it can't be, I don't know. But it was the reason why I did it like this.

Reply:
However, when the std::numeric_limits<float>::is_iec559 is true, it probably guarantees that sizeof(float)==4

Reply:
Sigh. I already said that you're "**not testing for it to be 32 bit or 4 bytes**". Now you say you do it because you want to test whether it's 4 bytes. This sounds like you didn't read my comment.

Reply:
> However, when the std::numeric_limits::is_iec559 is true, it probably guarantees that sizeof(float)==4

"Probably" isn't good enough, even less so if you're wrong.